litestar/contrib/repository/testing/generic_mock_repository.py

```diff
@@ -378,3 +379,358 @@ def seed_collection(cls, instances: Iterable[ModelT]) -> None:
+
+            return existing, False
+        return self.add(self.model_type(**kwargs)), True  # pyright: ignore[reportGeneralTypeIssues]
+
+    def get_one(self, **kwargs: Any) -> ModelT:
+        """Get instance identified by query filters.
+
+        Args:
+            **kwargs: Instance attribute value filters.
+
+        Returns:
+            The retrieved instance or None
+
+        Raises:
+            NotFoundError: If no instance found identified by ``kwargs``.
+        """
+        data = self.list(**kwargs)
+        return self.check_not_found(data[0] if data else None)
+
+    def get_one_or_none(self, **kwargs: Any) -> ModelT | None:
+        """Get instance identified by query filters or None if not found.
+
+        Args:
+            **kwargs: Instance attribute value filters.
+
+        Returns:
+            The retrieved instance or None
+        """
+        data = self.list(**kwargs)
+        return data[0] if data else None
+
+    def count(self, *filters: FilterTypes, **kwargs: Any) -> int:
+        """Count of rows returned by query.
+
+        Args:
+            *filters: Types for specific filtering operations.
+            **kwargs: Instance attribute value filters.
+
+        Returns:
+            Count of instances in collection, ignoring pagination.
+        """
+        return len(self.list(*filters, **kwargs))
+
+    def update(self, data: ModelT) -> ModelT:
+        """Update instance with the attribute values present on ``data``.
+
+        Args:
+            data: An instance that should have a value for :attr:`id_attribute <AsyncGenericMockRepository.id_attribute>` that exists in the
+                collection.
+
+        Returns:
+            The updated instance.
+
+        Raises:
+            NotFoundError: If no instance found with same identifier as ``data``.
+        """
+        item = self._find_or_raise_not_found(self.get_id_attribute_value(data))
+        self._update_audit_attributes(data, do_created=False)
+        for key, val in data.__dict__.items():
+            if key.startswith("_"):
+                continue
+            setattr(item, key, val)
+        return item
+
+    def update_many(self, data: list[ModelT]) -> list[ModelT]:
+        """Update instances with the attribute values present on ``data``.
+
+        Args:
+            data: A list of instances that should have a value for :attr:`id_attribute <AsyncGenericMockRepository.id_attribute>`
+                that exists in the collection.
+
+        Returns:
+            The updated instances.
+
+        Raises:
+            NotFoundError: If no instance found with same identifier as ``data``.
+        """
+        items = [self._find_or_raise_not_found(self.get_id_attribute_value(row)) for row in data]
+        now = self._now()
+        for item in items:
+            self._update_audit_attributes(item, do_created=False, now=now)
+            for key, val in item.__dict__.items():
+                if key.startswith("_"):
```

Comment:
I think you can create a utility for all the places you iterate like this, `__dict__` and return the fields from it:

```python
def model_items(model: ModelT) -> list[tuple[Str, Any]]:
   return [(k,v) for k,v in model.__dict__.items() if not k.startswith("_")]
```